getting-started/kubernetes/vpp/getting-started.md

```diff
@@ -0,0 +1,199 @@
+
+The VPP dataplane is entirely compatible with the other {{site.prodname}} dataplanes, meaning you can have a cluster with VPP-enabled nodes along with regular nodes. This makes it easy to migrate a cluster from Linux or eBPF networking to VPP networking.
+
+In the future, the VPP dataplane will offer additional features for network-intensive applications, such as providing `memif` userspace packet interfaces to the pods (instead of regular Linux network devices).
+
+Trying out the tech preview will give you a taste of these benefits and an opportunity to give feedback to the VPP dataplane team.
+
+
+### Features
+
+This how-to guide uses the following {{site.prodname}} features:
+
+- **calico/node**
+- **VPP dataplane**
+
+### Concepts
+
+#### VPP
+
+The Vector Packet Processor (VPP) is a high-performance, open-source userspace network dataplane written in C, developed under the [fd.io](https://fd.io) umbrella. It supports many standard networking features (L2, L3 routing, NAT, encapsulations), and is easily extensible using plugins. The VPP dataplane uses plugins to efficiently implement Kubernetss services load balancing and {{site.prodname}} policies.
+
+### How to
+
+This guide details two ways to install Calico with the VPP dataplane:
+- On a [managed EKS cluster](#install-calico-with-the-vpp-dataplane-on-an-eks-cluster). This is the option that requires the least configuration.
+- On [any Kubernetes cluster](#install-calico-with-the-vpp-dataplane-on-any-kubernetes-cluster).
+
+In both cases, here are the details of what you will get:
+
+{% include geek-details.html details='Policy:Calico,IPAM:Calico,CNI:Calico,Overlay:IPIP,Routing:BGP,Datastore:Kubernetes' %}
+
+### Install Calico with the VPP dataplane on an EKS cluster
+
+#### Requirements
+
+For these instructions, we will use `eksctl` to provision the cluster. However, you can use any of the methods in {% include open-new-window.html text='Getting Started with Amazon EKS' url='https://docs.aws.amazon.com/eks/latest/userguide/getting-started.html' %}
+
+Before you get started, make sure you have downloaded and configured the {% include open-new-window.html text='necessary prerequisites' url='https://docs.aws.amazon.com/eks/latest/userguide/getting-started-eksctl.html#eksctl-prereqs' %}
+
+#### Provision and configure the cluster
+
+1. First, create an Amazon EKS cluster without any nodes.
+
+   ```bash
+   eksctl create cluster --name my-calico-cluster --without-nodegroup
+   ```
+
+2. Since this cluster will use {{site.prodname}} for networking, you must delete the `aws-node` daemon set to disable AWS VPC networking for pods.
+
+   ```bash
+   kubectl delete daemonset -n kube-system aws-node
+   ```
+
+3. Now that you have a cluster configured, you can install {{site.prodname}}.
+
+   ```bash
+   kubectl apply -f https://raw.githubusercontent.com/projectcalico/vpp-dataplane/master/yaml/generated/calico-vpp-eks.yaml
```

Comment:
We can keep it as a single manifest for now, but I'd like to see this split eventually so that the VPP data plane resources are a distinct addon.

e.g.,

```
# Install the VPP data plane
kubectl create -f https://docs.projectcalico.org/manifests/vpp-dataplane.yaml
```

```
# Install Calico control plane with VPP customizations.
kubectl create -f https://docs.projectcalico.org/manifests/calico.yaml -k <kustomization_directory>
```

It makes it really clear which components belong to which subsystem (similar to my related comment on configmap name).

This will also be important once we move to support the operator as well, since we will want the vpp-dataplane.yaml step to apply to both operator and manifest installs without needing to maintain multiple copies of the same resources in different manifest files.

Reply:
Similarly, with the operator we're trying to follow a better practice in terms of grouping resources in appropriate namespaces. As part of that we're moving most (if not everything) out of kube-system moving forward. For example, the new API server component I added for v3.19 goes in the `calico-apiserver` namespace.

It provides better RBAC controls, limits the blast-radius for mistakes, makes it easier to write policies, etc. 

That said...

```yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  labels:
    k8s-app: calico-vpp-node
  name: calico-vpp-node
  namespace: kube-system
```

Feels like there is a case for doing the same for the VPP data plane components. So, the above would become something like this:

```yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  labels:
    k8s-app: vpp-dataplane
  name: vpp-dataplane
  namespace: calico-vpp-dataplane
```

We can probably wait until after tech-preview to do this if it's too burdensome. The benefit to doing this prior is that it will be a less disruptive change for tech-preview users, but we don't guarantee a lack of disruption anyway so this is OK.

Sorry for not spotting this one earlier - let me know if you have any questions!